src/Containers/App/index.jsx

```diff
@@ -777,7 +777,36 @@ class App extends Component {
     }
 
     try {
-      const apiVersion = await this.serial.getApiVersion();
+      const apiVersion = await this.serial.getApiVersion().catch(async (e) => {
+        if (e.message.includes("Timed out after")) {
+          let hasResets = false;
+          let i = 0;
+          let esc = await this.serial.getFourWayInterfaceInfo(i).catch(() => null);
+          while (esc !== null) {
```

Comment:
Technically the assignment could be made in the while statement, this way you would not need to make it at the end of the while statement too.

Generally I am not too happy with the chained catch here either. Exit condition could simply be the exception.

Reply:
changed the while loop